### PR TITLE
Chip SHELL mbed integration

### PR DESCRIPTION
**Pull request by ATmobica**
#### Problem
CHIP project does not support mbed platform and does not include the example application

 #### Summary of Changes
Add shell example application for mbed platform
Add mbed platform support for CHIP project - add stubs and update the mbed build file
Create mbed config directory - add CMakeLists which prepares CHIP build configuration and defines 'chip-gn' target that builds CHIP library and defines 'chip' target that exposes CHIP headers & libraries to the application; add CHIP GN "project" to build GN library
Add mbed example build script
Create vscode task to build mbed shell example

### CODE REVIEW

**Review by fkjagodzinski — APPROVED** [review]
Great stuff! I left a few minor nitpicks to consider.

File: examples/shell/mbed/mbed-os.lib [hunk 1]
```diff
@@ -0,0 +1 @@
+https://github.com/ARMmbed/mbed-os#master
```
[the remark this answers is not in the document]
Comment (fkjagodzinski) [reply examples/shell/mbed/mbed-os.lib:1]: Ahh, we do not use it anyway. Forgot about that.

File: config/mbed/CMakeLists.txt [hunk 2]
```diff
@@ -132,34 +138,31 @@ else()
     set(CHIP_PROJECT_CONFIG "")
 endif()
 
+if (${CMAKE_BUILD_TYPE} STREQUAL "debug")
+    set(CONFIG_DEBUG "y")
```
Comment (pan-) [inline config/mbed/CMakeLists.txt:154]: That's a tricky one, mbed has three configuration: 
- `debug` => compilation without optimization and assertions
- `develop` => Compilation with optimization and assertions 
- `release` => compilation with optimization and without assertions

File: config/mbed/mbed-util.cmake [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+
+#
+#   @file
+#     CMake utilities for managing and retrieving mbed build configuration
+#
+
+function(get_flags_for_lang lang input output)
```
Comment (pan-) [inline config/mbed/mbed-util.cmake:22]: More comments explaining what the function does and what the arguments are would be welcomed.

**Review by pan- — APPROVED** [review]
Thanks @ATmobica That looks good. 
Could you fix the rebase, something seems off.

File: config/mbed/CMakeLists.txt [hunk 4]
```diff
@@ -93,18 +93,24 @@ endmacro()
 # Prepare CHIP configuration based on the project configuration
 # ==============================================================================
 
+file(STRINGS ${CMAKE_CURRENT_BINARY_DIR}/config ConfigContents)
```
[the remark this answers is not in the document]
Comment (ATmobica) [reply config/mbed/CMakeLists.txt:100]: I do not think so. Just a simple parser for the config file. I will add a comment on what is this.

File: config/mbed/CMakeLists.txt (hunk 2 above)
Comment (ATmobica) [reply config/mbed/CMakeLists.txt:154]: Right, but CHIP has only two states is_debug on or off, which means development features are enabled or disabled. 
The question is how to treat the development state. If it uses the optimization, it is more suited to is_debug=off.

**Review by lukgni — APPROVED** [review]
lgtm!

**Pull request merged**